### PR TITLE
Usergrid 608

### PR DESCRIPTION
**Pull request by GERey**

### CODE REVIEW

File: stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java [hunk 1]
```diff
@@ -173,6 +184,161 @@ public CandidateResults search( final SearchEdge searchEdge, final SearchTypes s
...
+        }
+
+        try {
+            //Added For Graphite Metrics
+            Timer.Context timeSearch = searchTimer.time();
+
+            //set the timeout on the scroll cursor to 6 seconds and set the number of values returned per shard to 100.
+            //The settings for the scroll aren't tested and so we aren't sure what vlaues would be best in a production enviroment
+            //TODO: review this and make them not magic numbers when acking this PR.
+            searchResponse = srb.setScroll( new TimeValue( 6000 ) ).setSize( 100 ).execute().actionGet();
```
Comment (GERey) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java]: This some comments on. Not sure what values we want here.

File: stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java [hunk 2]
```diff
@@ -173,6 +184,161 @@ public CandidateResults search( final SearchEdge searchEdge, final SearchTypes s
...
+            
+
+            while(true){
+                //add search result hits to some sort of running tally of hits.
+                candidates = aggregateScrollResults( candidates, searchResponse );
+
+                SearchScrollRequestBuilder ssrb = searchRequestBuilderStrategyV2
+                    .getScrollBuilder( searchResponse.getScrollId() )
+                    .setScroll( new TimeValue( 6000 ) );
```
Comment (GERey) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java]: same as above

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/EventBuilderImpl.java [hunk 3]
```diff
@@ -125,12 +128,30 @@ public EntityDeleteResults queueEntityDelete( final ApplicationScope application
...
         final GraphManager gm = graphManagerFactory.createEdgeManager( applicationScope );
 
+        //needs get versions here.
+
+
+        //TODO: change this to be an observable
+        //so we get these versions and loop through them until we find the MvccLogEntry that is marked as delete.
+        //TODO: evauluate this to possibly be an observable to pass to the nextmethod.
+        MvccLogEntry mostRecentlyMarked = ecm.getVersions( entityId ).toBlocking()
+                                             .firstOrDefault( null,
```
Comment (GERey) [inline stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/EventBuilderImpl.java]: Should be combined with the observable below. There isn't any reasons we should go through entity versions twice.

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/EventBuilderImpl.java [hunk 4]
```diff
@@ -125,12 +128,30 @@ public EntityDeleteResults queueEntityDelete( final ApplicationScope application
...
+        //If there is nothing marked then we shouldn't return any results.
+        //TODO: evaluate if we want to return null or return empty observable when we don't have any results marked as deleted.
+        if(mostRecentlyMarked == null)
+            return null;
 
         //observable of index operation messages
+        //this method will need the most recent version.
+        //When we go to compact the graph make sure you turn on the debugging mode for the deleted nodes so
+        //we can verify that we mark them. That said that part seems kinda done. as we also delete the mvcc buffers.
         final Observable<IndexOperationMessage> edgeObservable =
```
Comment (shawnfeldman) [inline stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/EventBuilderImpl.java]: probably needs to be flatmap

File: stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java [hunk 5]
```diff
@@ -173,6 +184,159 @@ public CandidateResults search( final SearchEdge searchEdge, final SearchTypes s
...
+            //Added For Graphite Metrics
+            Timer.Context timeSearch = searchTimer.time();
+
+            //set the timeout on the scroll cursor to 6 seconds and set the number of values returned per shard to 100.
+            //The settings for the scroll aren't tested and so we aren't sure what vlaues would be best in a production enviroment
+            //TODO: review this and make them not magic numbers when acking this PR.
+            searchResponse = srb.setScroll( new TimeValue( 6000 ) ).setSize( 100 ).execute().actionGet();
+
+
+            while(true){
```
Comment (shawnfeldman) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java]: make this into observable

Comment (shawnfeldman) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/persistence/index/impl/EsApplicationEntityIndexImpl.java]: needs to be like IndexBufferConsumer

**Pull request merged**